core/src/mender-client.c

```diff
@@ -1153,12 +1135,8 @@ mender_client_update_work_function(void) {
         continue;                                                                                \
     }
 
-    {
-        cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
-        if (NULL != deployment_id_j) {
-            deployment_id = deployment_id_j->valuestring;
-        }
-    }
+    mender_deployment_data_get_id(mender_client_deployment_data, &deployment_id);
```

Comment:
`mender_client_deployment_data` can be NULL here, which will make the `assert(NULL != deployment_data)` in `__mender_deployment_data_get_string()` fail

Reply:
Added if guard around it. Not quite sure if it is a proper solution. At least it crashed for @danielskinstad before, and does not crash for him now.